src/main/java/net/fabricmc/loader/FabricLoader.java

```diff
@@ -242,9 +244,14 @@ public MappingResolver getMappingResolver() {
 		return Collections.unmodifiableList(mods);
 	}
 
+	@Override
+	public Collection<net.fabricmc.loader.api.ModContainer> getFakeMods() {
+		return Collections.unmodifiableList(mods);
+	}
+
 	@Override
 	public boolean isModLoaded(String id) {
-		return modMap.containsKey(id);
+		return modMap.containsKey(id) || fakeModMap.containsKey(id);
```

Comment:
My request for this would require a contract to be established that a `IllegalArgumentException` would be thrown if two ModContainers of the same modid were to be registered during the registration process.

Reply:
oh that's a good point.